Data/ByteString/Builder/RealFloat.hs

```diff
@@ -0,0 +1,273 @@
+    fp <- mallocByteString 25:: IO (ForeignPtr Word8)
+    withForeignPtr fp $ \p ->
+      PS fp 0 . fromIntegral <$> c_ryu_d2s_to_chars (fromIntegral m) (fromIntegral e) (asCBool s) p
+
+
+-- auxiliary fixed format printing functions
+decimalLength9:: Integral a => a -> Int
+decimalLength9 v
+  | v >= 100000000 = 9
+  | v >= 10000000 = 8
+  | v >= 1000000 = 7
+  | v >= 100000 = 6
+  | v >= 10000 = 5
+  | v >= 1000 = 4
+  | v >= 100 = 3
+  | v >= 10 = 2
+  | otherwise = 1
+
+decimalLength17 :: Integral a => a -> Int
+decimalLength17 v
+  | v >= 10000000000000000 = 17
+  | v >= 1000000000000000 = 16
+  | v >= 100000000000000 = 15
+  | v >= 10000000000000 = 14
+  | v >= 1000000000000 = 13
+  | v >= 100000000000 = 12
+  | v >= 10000000000 = 11
+  | v >= 1000000000 = 10
+  | v >= 100000000 = 9
+  | v >= 10000000 = 8
+  | v >= 1000000 = 7
+  | v >= 100000 = 6
+  | v >= 10000 = 5
+  | v >= 1000 = 4
+  | v >= 100 = 3
+  | v >= 10 = 2
+  | otherwise = 1
+
+sign :: RealFloat a => a -> Builder
+sign f = if f < 0 then char7 '-' else mempty
+
+specialStr :: RealFloat a => a -> Maybe Builder
+specialStr f
+  | isNaN f          = Just $ string7 "NaN"
+  | isInfinite f     = Just $ sign f `mappend` string7 "Infinity"
+  | isNegativeZero f = Just $ string7 "-0.0"
+  | f == 0           = Just $ string7 "0.0"
+  | otherwise        = Nothing
+
+-- show fixed floating point matching show / showFFloat output by dropping
+-- digits after exponentiation precision
+ryu_d2fixed :: Double -> Maybe Int -> Builder
+ryu_d2fixed f prec
+  | Just b <- specialStr f = b
+  | otherwise = sign f `mappend` showFixed m e' prec
+  where (FD64 m e) = ryu_d2s_fd f
+        olength = decimalLength17 m
+        -- NB: exponent in exponential format is e' - 1
+        e' = fromIntegral e + olength
+
+showFixed :: Word64 -> Int -> Maybe Int -> Builder
+showFixed m e prec =
+    case prec of
+       Nothing
+         | e <= 0 -> char7 '0'
+                  `mappend` char7 '.'
+                  `mappend` string7 (replicate (-e) '0')
+                  `mappend` foldr mappend mempty (toB ds)
+         | otherwise ->
+           let f 0 s     rs = mk0 (reverse s) `mappend` char7 '.' `mappend` mk0 rs
+               f n s     [] = f (n-1) (char7 '0':s) []
+               f n s (r:rs) = f (n-1) (r:s) rs
+            in f e [] (toB ds)
+       Just p
+         | e >= 0 ->
+           let (ei, is') = roundTo 10 (p' + e) ds
+               (ls, rs) = splitAt (e + ei) (toB is')
+            in mk0 ls `mappend` mkDot rs
+         | otherwise ->
+           let (ei, is') = roundTo 10 p' (replicate (-e) 0 ++ ds)
+               (b:bs) = toB (if ei > 0 then is' else 0:is')
+            in b `mappend` mkDot bs
+           where p' = max p 0
+    where
+        mk0 ls = case ls of [] -> char7 '0'; _ -> foldr mappend mempty ls
+        mkDot rs = if null rs then mempty else char7 '.' `mappend` foldr mappend mempty rs
+        ds = digits m
+        toB = fmap (char7 . intToDigit)
```

Comment:
The name `toB` is used elsewhere (internally) in `bytestring` as a short name for `liftFixedToBounded`, renaming this would be less confusing.

Reply:
Sure. https://github.com/haskell/bytestring/pull/222/commits/44336fc56ba0e4c317cc2fd6ac403b797f31e1e8